bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Button.java

```diff
@@ -1552,7 +1552,7 @@ private static void handleDPIChange(Widget widget, int newZoom, float scalingFac
 	}
 	// Refresh the image
 	if (button.image != null) {
-		button._setImage(Image.win32_new(button.image, newZoom));
+		button._setImage(button.image);
```

Comment:
Just for my understanding: Why set it to "what was already the current value" (is my assumption correct?) instead of using the zoom? Does this mean that the button now only has the image in the original size (scaling factor of `1`) and the consumers should rescale it?

Reply:
the setImage method calls the OS at the low level to set the image again which has some callbacks. Since the zoom level has changed, on the callback the handle extracted by win32_getHandle method gets the handle for the specified zoom level from the imge object. And the zoomed handle of the image is set for the button.